finagle-base-http/src/main/scala/com/twitter/finagle/http/CookieMap.scala

```diff
@@ -157,17 +162,21 @@ class CookieMap private[finagle] (message: Message, cookieCodec: CookieCodec)
   private[this] def addNoRewrite(name: String, cookie: Cookie): Unit = {
     val prev = underlying(name)
     val next =
-      if (prev.contains(cookie)) cookie :: prev.filter(c => c != cookie)
+      if (message.isResponse && prev.contains(cookie)) cookie :: prev.filter(c => c != cookie)
       else cookie :: prev
 
     underlying(name) = next
   }
 
   /**
-   * Adds the given `cookie` with `name` into this map. Existing cookies
-   * with this name but different domain/path will be kept. If there is already
-   * an identical cookie (different value but name/path/domain is the same) in the
-   * map, it will be replaced within a new version.
+   * Adds the given `cookie` with `name` into this map.
+   *
+   * On a Response: existing cookies with this name but different domain/path
```

Comment:
issue: this response vs request behavior is strange to me.  what's the motivation behind it?  I don't see anything in the RFC that would encourage it.

Reply:
In a request, the cookie header does not provide any information on the domain/path for the cookie, only the cookie name and value; yet the cookie header can contain multiple values for the same name (presumably because they were set with different domains/paths that can both match for this request, but we don't have this information when receiving the request on the server side).

Reply:
OK, that makes sense to me, thanks!  It might be good to explain that in the scaladoc.